src/lib.rs

```diff
@@ -494,17 +500,14 @@ pub fn make_multiproof(root: &Node, keys: Vec<NibbleKey>) -> Result<Multiproof,
                 .to_string());
             }
 
-            if *leafkey == keys[0] {
+            let rlp = if *leafkey == keys[0] {
                 instructions.push(Instruction::LEAF(leafkey.len()));
-                let rlp = rlp::encode(&Leaf(leafkey.clone(), leafval.clone()));
-                values.push(rlp);
+                rlp::encode(&Leaf(leafkey.clone(), leafval.clone()))
             } else {
-                return Err(format!(
-                    "Trying to apply the wrong key {:?} != {:?}",
-                    keys[0], leafkey
-                )
-                .to_string());
-            }
+                instructions.push(Instruction::LEAF(keys[0].len()));
+                rlp::encode(&Leaf(keys[0].clone(), vec![]))
```

Comment:
> yes, 11111111111111110000000000000000 should be present, and that's (part of) the bug.

Well here it could be (similar to above) `rlp::encode(&Leaf(leafkey.clone(), leafval.clone()))` to include 11..00..

Maybe this `if *leafkey == keys[0] {` condition is not necessary at all. Whatever the path, we include this leaf fully.